code/datums/outputs.dm

```diff
@@ -52,26 +51,28 @@ GLOBAL_LIST_EMPTY(outputs_list)
 				if(distance <= 1)
 					pressure_factor = max(pressure_factor, 0.15) //touching the source of the sound
 
-				S.volume *= pressure_factor
+				sound_output.volume *= pressure_factor
 				//End Atmosphere affecting sound
 
-			if(S.volume <= 0)
+			if(sound_output.volume <= 0)
 				return //No sound
 
 			var/dx = turf_source.x - T.x // Hearing from the right/left
-			S.x = dx
+			sound_output.x = dx
 			var/dz = turf_source.y - T.y // Hearing from infront/behind
-			S.z = dz
+			sound_output.z = dz
 			// The y value is for above your head, but there is no ceiling in 2d spessmens.
-			S.y = 1
-			S.falloff = (falloff ? falloff : FALLOFF_SOUNDS)
+			sound_output.y = 1
+			sound_output.falloff = (falloff ? falloff : FALLOFF_SOUNDS)
 
 	//Process icon
-	var/image/I = image(icon[1], turf_source, icon[2], icon[3])
-	if(S && vol)
-		I.alpha = I.alpha * (vol / 100)
+	if(stored_image)
+		stored_image.loc = turf_source
+		stored_image.alpha = 255 //reset alpha to 255 since we'll be reusing this image every time
+		if(sound_output && vol)
+			stored_image.alpha = stored_image.alpha * (vol / 100)
```

Comment:
As I said before you can move code around so you don't need to check sound_output here

Reply:
yes but i need to check if there's an image first and if i put that inside the sound check and there's no sound but an image, the image wont show

Reply:
The whole stored image block doesn't need to be in the sound_output block
Also stored_image should never be null with how you currently have this set up
```
stored_image.loc = turf_source
stored_image.alpha = 255 //reset alpha to 255 since we'll be reusing this image every time

if(sound_output)
	if(vol)
		stored_image.alpha = stored_image.alpha * (vol / 100)
	...[other sound stuff]
```

Reply:
ill tell you why this runtimes on discord